src/main/java/lotto/domain/Lottos.java

```diff
@@ -0,0 +1,50 @@
+package lotto.domain;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Lottos {
+
+    private final List<Lotto> lottos;
+
+    private Lottos(List<Lotto> lottos) {
+        this.lottos = lottos;
+    }
+
+    public static Lottos of(List<Lotto> lottos) {
+        return new Lottos(lottos);
+    }
+
+    public Map<Rank, Integer> countMatchesWith(WinningLotto winningLotto) {
+        Map<Rank, Integer> prizeCounts = initializeRankCounts();
+
+        lottos.forEach(lotto -> {
+            Rank rank = winningLotto.determineRank(lotto);
+            prizeCounts.put(rank, prizeCounts.get(rank) + 1);
+        });
+
+        return prizeCounts;
+    }
+
+    private Map<Rank, Integer> initializeRankCounts() {
+        return Arrays.stream(Rank.values())
+                .collect(Collectors.toMap(
+                        rank -> rank,
+                        rank -> 0,
+                        (a, b) -> a,
+                        () -> new EnumMap<>(Rank.class)
+                ));
```

Comment:
사소한 부분이긴 한데 stream문을 사용하는 것이 오히려 복잡해보일 수도 있어서, for문 사용도 고려해볼 수 있을 것 같습니다! (선택 사항인 것 같아요)

```suggestion
        Map<Rank, Integer> prizeCounts = new EnumMap<>(Prize.class);
        for (Rank rank : Rank.values()) {
            prizeCounts.put(rank, 0);
        }
        
        return prizeCounts
```